src/equisolve/utils/split_data.py

```diff
@@ -0,0 +1,431 @@
+        # for the train, test, and validation tensors, respectively
+        (
+            [
+                [in_train, in_test, in_val],
+                [out_train, out_test, out_val]
+            ]
+        ), grouped_labels = split_data(
+            tensors=[input, output],
+            axis="samples",
+            names=["structure"],
+            n_groups=3,  # for train-test-validation
+            group_sizes=[7, 2, 1],  # absolute; 7, 2, 1 for train, test, val
+            random_seed=100,
+        )
+        # Inspect the grouped structure indices
+        grouped_labels
+        >>> [
+            Labels(
+                [(3,), (7,), (1,), (8,), (0,), (9,), (2,)],
+                dtype=[('structure', '<i4')]
+            ),
+            Labels([(4,), (6,)], dtype=[('structure', '<i4')]),
+            Labels([(5,)], dtype=[('structure', '<i4')]),
+        ]
+    """
+    # Check input args and parse `tensors` and `names` into lists
+    tensors = [tensors] if isinstance(tensors, TensorMap) else tensors
+    names = [names] if isinstance(names, str) else names
+    _check_args(tensors, axis, names, n_groups, group_sizes, random_seed)
+
+    # Get array of unique indices to split by for each tensor in `tensors`
+    unique_idxs_list = [
+        equistore.unique_metadata(tensor, axis, names) for tensor in tensors
+    ]
+
+    # Check that the unique indices are equivalent for all input tensors
+    _check_labels_equivalent(unique_idxs_list)
+    unique_idxs = unique_idxs_list[0]
+
+    # Shuffle the unique indices according to the random seed
+    _shuffle_indices(unique_idxs, random_seed)
+
+    # Must be at least as many unique indices as groups
+    n_indices = len(unique_idxs)
+    if n_indices < n_groups:
+        raise ValueError(
+            f"the number of groups specified ({n_groups}) is greater than the"
+            + f" number of unique metadata indices ({n_indices}) for the"
+            + f" chosen axis {axis} and names {names}: {unique_idxs}"
+        )
+
+    # Get group sizes
+    group_sizes = _get_group_sizes(n_groups, len(unique_idxs), group_sizes)
+
+    # The sum of the absolute group sizes must be less than or equal to the
+    # number of unique indices
+    if n_indices < sum(group_sizes):
+        raise ValueError(
+            f"the sum of the absolute group sizes ({sum(group_sizes)}) is greater than "
+            + f"the number of unique metadata indices ({n_indices}) for the chosen "
+            + f"axis {axis} and names {names}: {unique_idxs}"
+        )
+
+    # Group the indices according to the group sizes
+    grouped_labels = _group_indices(unique_idxs, group_sizes)
+
+    # Split each of the input TensorMaps
+    split_tensors = []
+    for tensor in tensors:
+        split_tensors.append(equistore.split(tensor, axis, grouped_labels))
+
+    return split_tensors, grouped_labels
+
+
+def _shuffle_indices(indices: Labels, random_seed: Optional[int] = None):
```

Comment:
I would rather say `None` means the function is picking a random number and `-1` does not make sense. Why should one call a shuffle method if one doesn't want to shuffle the indices?

Also I would just call the parameter `seed`.

Reply:
Changed re: your suggestions; None doesn't shuffle, an int sets the seed for shuffling, -1 removed.